volume/service/service.go

```diff
@@ -61,11 +61,17 @@ func (s *VolumesService) GetDriverList() []string {
 //
 // A good example for a reference ID is a container's ID.
 // When whatever is going to reference this volume is removed the caller should defeference the volume by calling `Release`.
-func (s *VolumesService) Create(ctx context.Context, name, driverName string, opts ...opts.CreateOption) (*types.Volume, error) {
+func (s *VolumesService) Create(ctx context.Context, name, driverName string, options ...opts.CreateOption) (*types.Volume, error) {
 	if name == "" {
 		name = stringid.GenerateRandomID()
 	}
-	v, err := s.vs.Create(ctx, name, driverName, opts...)
+
+	oldV, err := s.Get(ctx, name, opts.WithGetDriver(driverName))
+	if err == nil {
+		return oldV, nil
+	}
+
+	v, err := s.vs.Create(ctx, name, driverName, options...)
```

Comment:
Perhaps instead we should make this return wether or not it created the volume, and send the right event based on that?